File: backend/models/message.go
package models

import "time"

type Message struct {
	ID         uint      `json:"id"`
	SenderID   uint      `gorm:"index;not null" json:"sender_id"`
	ReceiverID uint      `gorm:"index;not null" json:"receiver_id"`
	Content    string    `gorm:"not null" json:"content"`
	CreatedAt  time.Time `json:"timestamp"`
}

File: OpenMarket/Models/Message.swift
import Foundation

struct Message: Codable, Identifiable {
    let id: Int
    let senderID: Int
    let receiverID: Int
    let content: String
    let timestamp: Date

    enum CodingKeys: String, CodingKey {
        case id, content, timestamp
        case senderID   = "sender_id"
        case receiverID = "receiver_id"
    }
}

struct SendMessageRequest: Encodable {
    let receiverID: Int
    let content: String

    enum CodingKeys: String, CodingKey {
        case receiverID = "receiver_id"
        case content
    }
}
